### PR TITLE
convert scripts to console_scripts entry_points

### PR DESCRIPTION
**Pull request by jhkennedy**
:wave: I maintain the [conda-forge recipe](https://github.com/conda-forge/remotezip-feedstock) for `remotezip` and this PR is to support that, and some users downstream (e.g., https://github.com/asfadmin/Discovery-asf_search/issues/240)

Over on the conda side of the house, we can build pure python packages like remotezip as "noarch" packages to get maximum build portability. But, there are a [few constraints](https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python), that prevent `remotezip` from being distributed as a noarch package.

Specifically, `remotezip` using `scripts` in `setup.py` prevents a noarch build:
https://github.com/gtsystem/python-remotezip/blob/master/setup.py#L19

largely because it's not as portable, doesn't work well on Windows, and is [not recommended by the Python Packaging Authority](https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#scripts):
>  Although setup() supports a [scripts](https://setuptools.pypa.io/en/latest/deprecated/distutils/setupscript.html#distutils-installing-scripts) keyword for pointing to pre-made scripts to install, the recommended approach to achieve cross-platform compatibility is to use [console_scripts](https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#console-scripts) entry points

[`console_script` `entry_points`](https://packaging.python.org/en/latest/specifications/entry-points/) are well supported on the conda-forge side, and since the `bin/remotezip` script is a python script, it can be easily converted. 

**This PR** converts the `remotezip` CLI from a python script to a `console_script` `entry_points`, making `remotezip` a single-file "package". 

I would not expect this to be noticeable by users as `pip` will still install a `remotezip` script with the same CLI interface.

### CODE REVIEW

File: remotezip.py [hunk 1]
```diff
@@ -252,4 +255,45 @@ def _get_position_to_size(self):
         return {a: b-a for a, b in pairwise(ilist)}
 
 
-
+def list_files(url, support_suffix_range, filenames):
```
Comment (gtsystem) [inline remotezip.py:258]: I would prefer to keep this function and the next private as they are not really part of the module. Can you prefix them with underscore?

Comment (jhkennedy) [reply remotezip.py]: @gtsystem Yes, done! I went back and forth on it when I moved them.

File: setup.py
[the remark this answers is not in the document]
Comment (jhkennedy) [reply setup.py:1]: Done!

**Review by gtsystem — APPROVED** [review]

**Comment by gtsystem** [comment]
Tests are failing due to the new imports.
Can you add these dependencies in https://github.com/gtsystem/python-remotezip/blob/master/.github/workflows/python-package.yml#L28 ?

**Comment by jhkennedy** [comment]
> Tests are failing due to the new imports.
Can you add these dependencies in https://github.com/gtsystem/python-remotezip/blob/master/.github/workflows/python-package.yml#L28 ?

Oh, whoops. @gtsystem, it'd probably be better just to pip install `remotezip` from the repository checkout so that the runtime dependencies are picked up and the CLI script is available for testing (though I don't see it being tested). 

And since `tests_require` is a [deprecated setuptools keyword](https://setuptools.pypa.io/en/latest/references/keywords.html#keywords), moving the test dependencies to a `test` group in `extras_requires`. That would be applying these changes:
```diff
diff --git a/.github/workflows/python-package.yml b/.github/workflows/python-package.yml
index c56f2c7..1b1f64d 100644
diff --git a/setup.py b/setup.py
index efebbe5..c6a9551 100644
--- a/setup.py
+++ b/setup.py
@@ -15,7 +15,9 @@ setup(
     long_description=description,
     long_description_content_type="text/markdown",
     install_requires=["requests", "tabulate"],
-    tests_require=['requests_mock'],
+    extras_require={
+        "test": ["requests", "requests_mock"],
+    },
     entry_points={
         'console_scripts': ['remotezip = remotezip:main']
     },
--- a/.github/workflows/python-package.yml
+++ b/.github/workflows/python-package.yml
@@ -25,12 +25,13 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install requests requests_mock wheel
+        python -m pip install '.[test]'
     - name: Test
       run: |
         python test_remotezip.py -v
     - name: Package
       run: |
+        python -m pip install wheel
         python setup.py bdist_wheel
     - name: Archive artifact
       uses: actions/upload-artifact@v2
```
I've confirmed the tests will pass with the above changes. I can push those changes if you like?

**Comment by gtsystem** [comment]
> I've confirmed the tests will pass with the above changes. I can push those changes if you like?

Yes thanks

**Comment by jhkennedy** [comment]
Pushed!

**Comment by jhkennedy** [comment]
@gtsystem done! I also set [fast-fail](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast) to false so tests won't be canceled if one fails.

**Comment by gtsystem** [comment]
Merged, thanks again!

**Comment by jhkennedy** [comment]
Fantastic! Thanks for you help @gtsystem !

**Comment by jhkennedy** [comment]
@gtsystem can you push this up to PyPI? Is there anything else you'd like done first that I can help with?

**Comment by gtsystem** [comment]
@jhkennedy, sorry for the delay, just did. Can you check https://pypi.org/project/remotezip/0.12.2/ ?

**Comment by jhkennedy** [comment]
@gtsystem Looks good to me! I was able to pip install it, the entry point ran fine on the CLI and I could load in and use it from the python interpreter. 

Thank you very much for your help!

**Pull request merged**